packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -141,47 +164,43 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     self.result = nil;
   }
 
+  self.result = result;
+  _arguments = call.arguments;
+
   if ([@"pickImage" isEqualToString:call.method]) {
-    self.result = result;
-    _arguments = call.arguments;
-    int imageSource = [[_arguments objectForKey:@"source"] intValue];
+    int imageSource = [call.arguments[@"source"] intValue];
 
     if (imageSource == SOURCE_GALLERY) {  // Capture is not possible with PHPicker
       if (@available(iOS 14, *)) {
         // PHPicker is used
         [self pickImageWithPHPicker:1];
       } else {
         // UIImagePicker is used
-        [self pickImageWithUIImagePicker];
+        [self launchUIImagePickerWithSource:imageSource];
       }
     } else {
-      [self pickImageWithUIImagePicker];
+      [self launchUIImagePickerWithSource:imageSource];
     }
   } else if ([@"pickMultiImage" isEqualToString:call.method]) {
     if (@available(iOS 14, *)) {
-      self.result = result;
-      _arguments = call.arguments;
       [self pickImageWithPHPicker:0];
     } else {
-      [self pickImageWithUIImagePicker];
+      [self launchUIImagePickerWithSource:SOURCE_GALLERY];
     }
   } else if ([@"pickVideo" isEqualToString:call.method]) {
-    _imagePickerController = [[UIImagePickerController alloc] init];
-    _imagePickerController.modalPresentationStyle = UIModalPresentationCurrentContext;
-    _imagePickerController.delegate = self;
-    _imagePickerController.mediaTypes = @[
+    UIImagePickerController *imagePickerController = [self createImagePickerController];
```

Comment:
Shouldn't this be `self.imagePickerController`?

If we need a stateful controller, I'm very concerned about our test coverage if this didn't break anything. Does it not actually need to be stateful?

Reply:
With some refactoring it is not necessary to keep the `imagePickerController` as state. I did the refactoring necessary which involves passing the current copy of the `imagePickerController` around to some other methods. Please have a look if you agree with this.

Reply:
Looks great, thanks! (Now we just need to do the same with arguments, and ideally result, but those are definitely out of scope.)